util/warnp.c

```diff
@@ -112,6 +116,10 @@ void
 warnp_syslog(int enable)
 {
 
+	/* If we're currently using syslog, close the file descriptor. */
+	if (use_syslog)
```

Comment:
Maybe worth making this `use_syslog && !enable`?  No point closing a descriptor if we're about to use it again.

Reply:
Heh, I had that in the first place, then I figured that nobody would call `warnp_syslog(1)` twice so I might as well go with simpler code.

Done.